scipy/optimize/_differentialevolution.py

```diff
@@ -434,6 +439,13 @@ def init_population_random(self):
         rng = self.random_number_generator
         self.population = rng.random_sample(self.population_shape)
 
+        # reset population energies
+        self.population_energies = (np.ones(self.num_population_members)
+                                    * np.inf)
+
+        # reset number of function evaluations counter
+        self._nfev = 0
+
```

Comment:
Could you find a way to factor out what's similar between `init_population_*` methods?